packages/frontend-defaults/src/createApp.tsx

```diff
@@ -87,29 +94,93 @@ export function createApp(options?: CreateAppOptions): {
         overrideBaseUrlConfigs(defaultConfigLoaderSync()),
       );
 
-    const discoveredFeatures = getAvailableFeatures(config);
+    const loadedFeatures: FrontendFeature[] = [];
+
+    // Separate deprecated CreateAppFeatureLoader elements from the frontend features,
+    // and manage the deprecated elements first.
+    const [providedFeatures, createAppFeatureLoaders] = (
+      options?.features ?? []
+    ).reduce<[FrontendFeature[], CreateAppFeatureLoader[]]>(
+      (acc, item) => {
+        if ('$$type' in item) {
+          acc[0].push(item);
+        } else {
+          acc[1].push(item);
+        }
+        return acc;
+      },
+      [[], []],
+    );
```

Comment:
Bit of a nit, but I think a `let a; let b; for (...) { if (...) { a.push(...) } else { b.push(...) }` would be a bit easier to read here. Same thing below

Reply:
Fixed